lib/url_parser.js

```diff
@@ -49,7 +49,8 @@ module.exports = function(url, options, callback) {
     }
 
     let connectionStrings = addresses.map(function(address, i) {
-      if (i === 0) return `mongodb://${address.name}:${address.port}`;
+      let base = result.auth ? `mongodb://${result.auth}@` : `mongodb://`;
```

Comment:
Can you compute `base` outside of the loop so that you don't do it on every iteration?